### DISCUSSION TITLE
.

### DISCUSSION THREAD

**Discussion by lucassebe** [cypress-io/cypress#26509] [Questions and Help] [answered]
.

**Comment by portyu9** [comment, marked as answer]
Hi @lucassebe 

Firebase Hosting cannot run Cypress tests. It is a hosting/CDN service for your web application, not a CI runner that executes Node.js processes.

A typical setup is:

1. Deploy the application under test to Firebase Hosting.
2. Run Cypress from a CI service such as GitHub Actions, Cloud Build, GitLab CI, etc.
3. Point Cypress at the deployed Firebase URL.

For example:

```sh
npm ci
npx cypress run --config baseUrl=https://your-project.web.app
```

If you want the tests to run immediately after deployment, put both operations in the same CI workflow: deploy first, wait until the deployed application is reachable, and then execute `cypress run`.

The Firebase “Hosting Setup Complete” page only means Firebase Hosting was deployed successfully; it does not mean the Cypress project itself is running there.

Cypress Cloud can optionally record, parallelize, and report CI test runs, but it is not a general-purpose server on which your application or Cypress process is hosted.

Please mark this answer as accepted if it helped, thank you.
